tests/integration/test_integration.py

```diff
@@ -0,0 +1,106 @@
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+# Copyright 2019 The celery-dyrygent Authors. All rights reserved.
+
+import hashlib
+import re
+import uuid
+from collections import (
+    defaultdict,
+)
+
+import celery
+import pytest
+from app import (
+    order_task,
+    read_logs,
+    return_value_task,
+)
+
+from celery_dyrygent.workflows import (
+    Workflow,
+)
+
+
+@pytest.fixture(scope='function')
+def test_name():
+    return str(uuid.uuid4())
+
+
+class Helpers:
+    @staticmethod
+    def md5(value):
+        """Md5 check. The same function exists in worker."""
+        return hashlib.md5(str(value).encode()).hexdigest()
+
+    @staticmethod
+    def parse_logs(logs):
+        task_regex = r'app.(?:order|return_value)_task\[([\w-]+)\]: Test name: ([\w-]+) Test value: ([\w-]+)'
+        grouped_logs = defaultdict(list)
+
+        for log in logs.splitlines():
+            findings = re.search(task_regex, log)
+            if findings:
+                _, test_name, test_value = findings.groups()
+                grouped_logs[test_name].append(test_value)
+
+        return grouped_logs
+
+
+class TestCeleryDyrygentReturnValue(Helpers):
+    def test_return_value(self, test_name):
+        canvas = return_value_task.s(test_name)
+
+        workflow = Workflow()
+        workflow.add_celery_canvas(canvas)
+        workflow.apply_async().wait()
+
+        task_id = list(workflow.nodes.keys())[0]
+        logs_result = read_logs.delay()
+        results = self.parse_logs(logs_result.get())[test_name]
+
+        assert self.md5(task_id) == results[0]
```

Comment:
Why do you use md5 here? Can't we just compare unhashed strings?
The readability is a bit blurry :racehorse:

Reply:
removed